sql/mysql_db/mysql_db_serialize.go

```diff
@@ -0,0 +1,192 @@
+		name := b.CreateString(column.Name())
+		privs := serializePrivilegeTypes(b, serial.PrivilegeSetColumnStartPrivsVector, column.ToSlice())
+
+		serial.PrivilegeSetColumnStart(b)
+		serial.PrivilegeSetColumnAddName(b, name)
+		serial.PrivilegeSetColumnAddPrivs(b, privs)
+		offsets[len(offsets)-i-1] = serial.PrivilegeSetColumnEnd(b) // reverse order
+	}
+	// Write column offsets (already reversed)
+	return serializeVectorOffsets(b, serial.PrivilegeSetTableStartColumnsVector, offsets)
+}
+
+func serializeTables(b *flatbuffers.Builder, tables []PrivilegeSetTable) flatbuffers.UOffsetT {
+	// Write table variables, and save offsets
+	offsets := make([]flatbuffers.UOffsetT, len(tables))
+	for i, table := range tables {
+		name := b.CreateString(table.Name())
+		privs := serializePrivilegeTypes(b, serial.PrivilegeSetTableStartPrivsVector, table.ToSlice())
+		cols := serializeColumns(b, table.GetColumns())
+
+		serial.PrivilegeSetTableStart(b)
+		serial.PrivilegeSetTableAddName(b, name)
+		serial.PrivilegeSetTableAddPrivs(b, privs)
+		serial.PrivilegeSetTableAddColumns(b, cols)
+		offsets[len(offsets)-i-1] = serial.PrivilegeSetTableEnd(b) // reverse order
+	}
+	// Write table offsets (order already reversed)
+	return serializeVectorOffsets(b, serial.PrivilegeSetDatabaseStartTablesVector, offsets)
+}
+
+// serializeDatabases writes the given Privilege Set Databases into the flatbuffer Builder, and returns the offset
+func serializeDatabases(b *flatbuffers.Builder, databases []PrivilegeSetDatabase) flatbuffers.UOffsetT {
+	// Write database variables, and save offsets
+	offsets := make([]flatbuffers.UOffsetT, len(databases))
+	for i, database := range databases {
+		name := b.CreateString(database.Name())
+		privs := serializePrivilegeTypes(b, serial.PrivilegeSetDatabaseStartPrivsVector, database.ToSlice())
+		tables := serializeTables(b, database.GetTables())
+
+		serial.PrivilegeSetDatabaseStart(b)
+		serial.PrivilegeSetDatabaseAddName(b, name)
+		serial.PrivilegeSetDatabaseAddPrivs(b, privs)
+		serial.PrivilegeSetDatabaseAddTables(b, tables)
+		offsets[len(offsets)-i-1] = serial.PrivilegeSetDatabaseEnd(b)
+	}
+
+	// Write database offsets (order already reversed)
+	return serializeVectorOffsets(b, serial.PrivilegeSetStartDatabasesVector, offsets)
+}
+
+func serializePrivilegeSet(b *flatbuffers.Builder, ps *PrivilegeSet) flatbuffers.UOffsetT {
+	// Write privilege set variables, and save offsets
+	globalStatic := serializePrivilegeTypes(b, serial.PrivilegeSetStartGlobalStaticVector, ps.ToSlice())
+	globalDynamic := serializeGlobalDynamic(b, ps.globalDynamic)
+	databases := serializeDatabases(b, ps.GetDatabases())
+
+	// Write PrivilegeSet
+	serial.PrivilegeSetStart(b)
+	serial.PrivilegeSetAddGlobalStatic(b, globalStatic)
+	serial.PrivilegeSetAddGlobalDynamic(b, globalDynamic)
+	serial.PrivilegeSetAddDatabases(b, databases)
+	return serial.PrivilegeSetEnd(b)
+}
+
+func serializeAttributes(b *flatbuffers.Builder, attributes *string) flatbuffers.UOffsetT {
```

Comment:
Should probably leave a comment explaining why the zero offset is necessary, and also that this should always be called as a result